alembic/autogenerate/rewriter.py

```diff
@@ -89,9 +88,56 @@ def add_column_idx(context, revision, op):
         """
         wr = self.__class__.__new__(self.__class__)
         wr.__dict__.update(self.__dict__)
-        wr._chained = other
+        if not isinstance(other, RewriterBase):
+            other = RewriterBase(process_revision_directives=other)
+        final_writer = wr
+        while final_writer._chained:
+            final_writer = final_writer._chained
+        final_writer._chained = other
```

Comment:
**Federico Caselli** (CaselIT) wrote:

I'm not sure this is the best options, since this changes a chained rewriter if it's parted is chained again. I mean that


    # r0, r1, r2 are rewriters
    r3 = r1.chain(r2) # r1 and r2 unchanged. r3._chainedis r2
    r4 = r3.chain(r0) # r2 is changed 
  
Maybe it's easier to change `_chained` to be a tuple that starts as an empty tuple (haven't tried it but I think it should work)?
At that point `RewriterBase` is no longer needed, and we can do here
```py
wr._chained += (other,)
```

it's fine for other to be a callable or another rewriter, The call method could then loop over _chained and call all items in the tuple

View this in Gerrit at https://gerrit.sqlalchemy.org/c/sqlalchemy/alembic/+/4990

Reply:
PR updated, thanks for the suggestion.